src/lib.rs

```diff
@@ -824,13 +844,20 @@ impl Queue {
         if msg.nfmark_dirty {
             nlmsg.put_be32(NFQA_MARK as u16, msg.nfmark);
         }
+
+        if let Some(ref conntrack) = msg.ct {
+            if conntrack.mark_dirty {
+                nlmsg.put_be32(CTA_MARK as u16, conntrack.mark);
```

Comment:
I think it needs to be nested within NFQA_CT

Reply:
Indeed, I noticed the nested related function calls in libnetfilter_queue, but I've been unable to understand what it was about, I tried many things but I'm not sure what it would translate to in your library.

The following doesn't work:

```rust
        if let Some(ref conntrack) = msg.ct {
            if conntrack.mark_dirty {
                nfq_hdr_put(&mut nlmsg, NFQA_CT as u16, msg.id, false);
                nlmsg.put_be32(CTA_MARK as u16, conntrack.mark);
            }
        }
```

The following error is returned:

```
Os { code: 22, kind: InvalidInput, message: "Invalid argument" }
```

Reply:
libnetfilter_conntrack appears to be using libmnl, which [defines](https://github.com/fqrouter/libmnl/blob/cf813753e8738c781e33de955f9733a27c65229f/include/libmnl/libmnl.h#L135C2-L135C17) `MNL_TYPE_NESTED`, I will try to understand what the underlying structure actually sent to the kernel looks like